Source/WebCore/rendering/svg/SVGTextMetricsBuilder.cpp

```diff
@@ -102,88 +102,126 @@ void SVGTextMetricsBuilder::initializeMeasurementWithTextRenderer(RenderSVGInlin
 {
+    auto [valueListPosition, lastCharacter] = state;
     SVGTextLayoutAttributes* attributes = text.layoutAttributes();
     Vector<SVGTextMetrics>* textMetricsValues = &attributes->textMetricsValues();
-    if (data->processRenderer) {
-        if (data->allCharactersMap)
+    if (data.processRenderer) {
+        if (data.allCharactersMap)
             attributes->clear();
         else
-            textMetricsValues->clear();
+            textMetricsValues->resize(0);
     }
 
     initializeMeasurementWithTextRenderer(text);
+
+    auto& scaledFont = text.scaledFont();
     bool preserveWhiteSpace = text.style().whiteSpaceCollapse() == WhiteSpaceCollapse::Preserve;
-    int surrogatePairCharacters = 0;
+    if (m_canUseSimplifiedTextMeasuring && data.processRenderer) {
+        // If we are not specifying specific configuration for characters, data.allCharactersMap has only 1 entry for default case.
+        // This is extremely common, and that's why we crafted a fast path here.
+        // FIXME: For any cases, we are handling one character by one character in SVGTextMetrics. But many texts do not have
+        // characterDataMap. We should handle multiple characters in one SVGTextMetrics. This also makes RTL work.
+        // FIXME: This function is called even though width information is not changed at all. RenderSVGText / RenderSVGInlineText
+        // should track the potential changes to width etc. and invoke this function only when it is actually changed.
+        if (data.allCharactersMap && m_run.direction() == TextDirection::LTR && data.allCharactersMap->size() == 1) {
+            constexpr unsigned defaultPosition = 1;
+            ASSERT(data.allCharactersMap->contains(defaultPosition)); // "1" is the default value and always exists.
+            auto characterData = data.allCharactersMap->get(defaultPosition);
+
+            auto view = m_run.text();
+            unsigned length = view.length();
+            unsigned skippedCharacters = 0;
+            float scalingFactor = text.scalingFactor();
+            ASSERT(scalingFactor);
+            float scaledHeight = scaledFont.metricsOfPrimaryFont().height() / scalingFactor;
+
+            // m_canUseSimplifiedTextMeasuring ensures that this does not include surrogate pairs. So we do not need to consider about them.
+            for (unsigned i = 0; i < length; ++i) {
+                UChar currentCharacter = view.characterAt(i);
+                if (currentCharacter == space && !preserveWhiteSpace && (!lastCharacter || lastCharacter == space)) {
+                    if (data.processRenderer)
+                        textMetricsValues->append(SVGTextMetrics(SVGTextMetrics::SkippedSpaceMetrics));
+                    ++skippedCharacters;
+                    continue;
+                }
+
+                if ((valueListPosition + i - skippedCharacters + 1) == defaultPosition)
+                    attributes->characterDataMap().set(i + 1, WTFMove(characterData));
```

Comment:
At first I wondered whether we could move from `characterData` more than once in this loop, but we can't, so it's OK. Though `SVGCharacterData` is just a struct of floats, so doesn't benefit from a `WTFMove` anyway.